optuna/integration/allennlp.py

```diff
@@ -1,501 +1,3 @@
-        # for a study name, trial id, monitor, and storage in environment variables.
-        else:
-            environment_variables = _get_environment_variables_for_trial()
-            study_name = environment_variables["study_name"]
-            trial_id = environment_variables["trial_id"]
-            monitor = environment_variables["monitor"]
-            storage = environment_variables["storage"]
-
-            if study_name is None or trial_id is None or monitor is None or storage is None:
-                message = (
-                    "Fail to load study. Perhaps you attempt to use `AllenNLPPruningCallback`"
-                    " without `AllenNLPExecutor`. If you want to use a callback"
-                    " without an executor, you have to instantiate a callback with"
-                    "`trial` and `monitor. Please see the Optuna example: https://github.com/"
-                    "optuna/optuna-examples/tree/main/allennlp/allennlp_simple.py."
-                )
-                raise RuntimeError(message)
-
-            else:
-                # If `stoage` is empty despite `study_name`, `trial_id`,
-                # and `monitor` are not `None`, users attempt to use `AllenNLPPruningCallback`
-                # with `AllenNLPExecutor` and in-memory storage.
-                # `AllenNLPruningCallback` needs RDB or Redis storages to work.
-                if storage == "":
-                    message = (
-                        "If you want to use AllenNLPExecutor and AllenNLPPruningCallback,"
-                        " you have to use RDB or Redis storage."
-                    )
-                    raise RuntimeError(message)
-
-                study = load_study(study_name, storage, pruner=_create_pruner())
-                self._trial = Trial(study, int(trial_id))
-                self._monitor = monitor
-
-    def on_epoch(
-        self,
-        trainer: "GradientDescentTrainer",
-        metrics: Dict[str, Any],
-        epoch: int,
-        is_primary: bool = True,
-        **kwargs: Any,
-    ) -> None:
-        """Check if a training reaches saturation.
-
-        Args:
-            trainer:
-                AllenNLP's trainer
-            metrics:
-                Dictionary of metrics.
-            epoch:
-                Number of current epoch.
-            is_primary:
-                A flag for AllenNLP internal.
-
-        """
-        if not is_primary:
-            return None
-
-        value = metrics.get(self._monitor)
-        if value is None:
-            return
-
-        self._trial.report(float(value), epoch)
-        if self._trial.should_prune():
-            raise optuna.TrialPruned()
+from optuna.integration._allennlp.dump_best_config import dump_best_config  # NOQA
+from optuna.integration._allennlp.executor import AllenNLPExecutor  # NOQA
+from optuna.integration._allennlp.pruner import AllenNLPPruningCallback  # NOQA
```

Comment:
I'm not sure, but should we leave this file? Could you tell me why we keep this file? Since we add all public classes to `optuna/integration/__init__.py`, we cannot have to keep this file.

Reply:
I assume that users do not access each integration class via the file/directory in `optuna/integration`.

Reply:
In order to provide `dump_best_config` under the namespace `optuna.integration.allennlp`, I'm wondering we have to leave this file. (pruner and executor are removed in cd890a1.

Reply:
Thank you for your reply. To be honest, I am not comfortable with the existence of the `allennlp.py` file and the `_allennlp` directory in the optuna/integration directory. In Optuna's naming conventions for files and directories, if they are not accessed by the user, they are preceded by a "_". Since the `dump_best_config` function is intended to be touched directly by the user from the outside, I would rename the `optuna/integration/_allennlp` directory itself to `optuna/integration/allennlp`. I think it is natural. As a result, the `optuna/integration/allennlpl.py` file will be deleted. 

I would like to hear other reviewers' opinions as well. @toshihikoyanase What do you think?

Reply:
I agree with you. `integration/allennlp/` also looks natural for me.

Reply:
Sorry for the delayed response. Creating `allennlp` directory in the `oputna/integration` looks great for me.

By the way, do users access the modules in `optuna.integration.allennlp`? For example, the current implementation allows users to access `optuna.integration.allennlp.pruner.AllenNLPPruningCallback`. I guess we can prefix  `_` to `pruner.py`, `environment.py` and so on.

Reply:
I'm sorry that I missed your suggestion.
It looks reasonable and applied it in 09cb153.